--- qx-build/source/class/qx/ui/form/TextField.js	2008-12-08 11:41:57.000000000 +0100
+++ qx-build/source/class/qx/ui/form/TextField.js	2008-12-08 15:58:48.000000000 +0100
@@ -71,6 +71,10 @@
     this.addEventListener("blur", this._onblur);
     this.addEventListener("focus", this._onfocus);
     this.addEventListener("input", this._oninput);
+    // [rst] Catch backspace in readonly text fields to prevent browser default
+    // action (which commonly triggers a history step back)
+    // See https://bugs.eclipse.org/bugs/show_bug.cgi?id=178320
+    this.addEventListener("keydown", this._onkeydown);
   },
 
 
@@ -927,6 +931,14 @@
       this.setValue(vValue);
     },
 
+    // [rst] Catch backspace in readonly text fields to prevent browser default
+    // action (which commonly triggers a history step back)
+    // See https://bugs.eclipse.org/bugs/show_bug.cgi?id=178320
+    _onkeydown : function( e ) {
+      if( e.getKeyIdentifier() == "Backspace" && this.getReadOnly() ) {
+        e.preventDefault();
+      }
+    },
 
     /*
     ---------------------------------------------------------------------------
