--- qx-build/source/class/qx/ui/menu/ButtonLayoutImpl.js	7 Aug 2007 16:30:44 -0000	1.1
+++ qx-build/source/class/qx/ui/menu/ButtonLayoutImpl.js	17 Sep 2007 13:20:22 -0000
@@ -197,7 +197,8 @@
           break;
 
         case vWidget._arrowObject:
-          vPos = vMenu.getArrowPosition();
+//          vPos = vMenu.getArrowPosition();
+          vPos = vMenu.getWidth() - 18;
           break;
       }
 
