--- qx-build/source/class/qx/core/Property.js	2008-12-08 11:41:57.000000000 +0100
+++ qx-build/source/class/qx/core/Property.js	2008-12-08 15:40:44.000000000 +0100
@@ -803,7 +803,8 @@
         }
 
         // Undefined check
-        if (variant === "set") {
+        // TODO [rh] unused: changed as in patch to http://bugzilla.qooxdoo.org/show_bug.cgi?id=599
+        if (qx.core.Variant.isSet("qx.debug", "on") && variant === "set") {
           code.push('if(value===undefined)prop.error(this,2,"'+name+'","'+variant+'",value);');
         }
       }
@@ -851,9 +852,10 @@
         code.push('var inherit=prop.$$inherit;');
       }
 
-      // Enable checks in debugging mode or then generating the setter
+      // Generate checks only in debug mode
 
-      if (incomingValue && (qx.core.Variant.isSet("qx.debug", "on") || variant === "set"))
+      // TODO [rh] unused: changed as in patch to http://bugzilla.qooxdoo.org/show_bug.cgi?id=599
+      if (incomingValue && qx.core.Variant.isSet("qx.debug", "on"))
       {
         // Null check
         if (!config.nullable) {
