--- qx-build/source/class/qx/ui/form/TextField.js	18 May 2009 09:44:03 -0000	1.4
+++ qx-build/source/class/qx/ui/form/TextField.js	25 May 2009 08:37:30 -0000
@@ -1371,7 +1371,12 @@
 
       // to be sure we get the element selected
       this._inputElement.select();
-      this._inputElement.focus();
+      // RAP [if] focus() leads to error in IE if the _inputElement is disabled.
+      // 277444: JavaScript error in IE when using setSelection on a ComboViewer with setEnabled is false
+      // https://bugs.eclipse.org/bugs/show_bug.cgi?id=277444
+      if( this.isEnabled() ) {
+        this._inputElement.focus();
+      }
     },
 
 
