--- qx-build/source/class/qx/ui/core/Widget.js	10 Dec 2008 16:57:54 -0000	1.1
+++ qx-build/source/class/qx/ui/core/Widget.js	2 Jul 2009 18:36:29 -0000
@@ -6891,6 +6891,13 @@
               es[i] = "";
           }
         }
+        
+        // [if] Fix for bug
+        // 279800: Some focused widgets look strange in webkit
+        // https://bugs.eclipse.org/bugs/show_bug.cgi?id=279800
+        if( qx.core.Variant.isSet( "qx.client", "webkit" ) ) {
+          es.outline = "none";
+        }
 
         for (var i in this._htmlProperties)
         {
