### Eclipse Workspace Patch 1.0
#P org.eclipse.rap.qx.build
Index: source/class/qx/event/handler/EventHandler.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/event/handler/EventHandler.js,v
retrieving revision 1.3
diff -u -r1.3 EventHandler.js
--- qx-build/source/class/qx/event/handler/EventHandler.js	30 Sep 2009 12:28:45 -0000	1.3
+++ qx-build/source/class/qx/event/handler/EventHandler.js	8 Oct 2009 12:04:57 -0000
@@ -22,7 +22,6 @@
 
 #module(ui_core)
 #optional(qx.event.handler.DragAndDropHandler)
-#optional(qx.ui.menu.Manager)
 #optional(qx.event.handler.FocusHandler)
 #optional(qx.ui.popup.PopupManager)
 #optional(qx.ui.popup.ToolTipManager)
@@ -398,6 +397,16 @@
     setAllowContextMenu : function( fun ) {
       this._allowContextMenu = fun;
     },
+
+    _menuManager : null,
+    
+    setMenuManager : function( manager ) {
+      this._menuManager = manager;
+    },
+    
+    getMenuManager : function( manager ) {
+      return this._menuManager; 
+    },
     
     /*
     ---------------------------------------------------------------------------
@@ -688,8 +697,8 @@
         {
           case "Escape":
           case "Tab":
-            if (qx.Class.isDefined("qx.ui.menu.Manager")) {
-              qx.ui.menu.Manager.getInstance().update(vTarget, vType);
+            if ( this._menuManager != null ) {
+              this._menuManager.update(vTarget, vType);
             }
 
             break;
@@ -1054,8 +1063,8 @@
             qx.ui.popup.PopupManager.getInstance().update(vTarget);
           }
 
-          if (qx.Class.isDefined("qx.ui.menu.Manager")) {
-            qx.ui.menu.Manager.getInstance().update(vTarget, vType);
+          if ( this._menuManager != null ) {
+            this._menuManager.update(vTarget, vType);
           }
 
           if (qx.Class.isDefined("qx.ui.embed.IframeManager")) {
@@ -1066,8 +1075,8 @@
 
         case "mouseup":
           // Mouseup event should always hide, independed of target, so don't send a target
-          if (qx.Class.isDefined("qx.ui.menu.Manager")) {
-            qx.ui.menu.Manager.getInstance().update(vTarget, vType);
+          if ( this._menuManager != null ) {
+            this._menuManager.update(vTarget, vType);
           }
 
           if (qx.Class.isDefined("qx.ui.embed.IframeManager")) {
@@ -1265,8 +1274,8 @@
       }
 
       // Hide Menus
-      if (qx.Class.isDefined("qx.ui.menu.Manager")) {
-        qx.ui.menu.Manager.getInstance().update();
+      if ( this._menuManager ) {
+        this._menuManager.update();
       }
 
       // Cancel Drag Operations
Index: source/class/qx/ui/popup/Popup.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/ui/popup/Popup.js,v
retrieving revision 1.1
diff -u -r1.1 Popup.js
--- qx-build/source/class/qx/ui/popup/Popup.js	10 Dec 2008 16:57:54 -0000	1.1
+++ qx-build/source/class/qx/ui/popup/Popup.js	8 Oct 2009 12:04:57 -0000
@@ -410,9 +410,10 @@
     {
       var vPopups = qx.lang.Object.getValues(qx.ui.popup.PopupManager.getInstance().getAll());
 
-      if (qx.Class.isDefined("qx.ui.menu.Manager"))
+      if (qx.event.handler.EventHandler.getInstance().getMenuManager() != null)
       {
-        var vMenus = qx.lang.Object.getValues(qx.ui.menu.Manager.getInstance().getAll());
+        var manager = qx.event.handler.EventHandler.getInstance().getMenuManager();
+        var vMenus = qx.lang.Object.getValues(manager.getAll());
         var vAll = vPopups.concat(vMenus).sort(qx.util.Compare.byZIndex);
       }
       else
