--- qx-build/source/class/qx/event/handler/EventHandler.js	30 Sep 2009 12:28:45 -0000	1.3
+++ qx-build/source/class/qx/event/handler/EventHandler.js	8 Oct 2009 13:22:15 -0000
@@ -989,7 +989,7 @@
         // Needed for Tooltips, ...
         qx.event.type.MouseEvent.storeEventState(vEventObject);
 
-        if (vTargetIsEnabled)
+        if (vDispatchTarget.getEnabled())
         {
           // Dispatch Event through target (eventtarget-)object
           var vEventWasProcessed = false;
