--- qx-build/source/class/qx/ui/form/TextField.js	2 Sep 2009 10:55:19 -0000	1.7
+++ qx-build/source/class/qx/ui/form/TextField.js	20 Oct 2009 09:56:39 -0000
@@ -868,7 +868,7 @@
 // RAP [if] Centered text field vertically
     _centerFieldVertically : function() {
       if( this._inputTag === "input" && this._inputElement ) {
-        var innerHeight = this.getInnerHeight();
+        var innerHeight = qx.html.Dimension.getAreaHeight( this.getElement() );
         var inputElementHeight = qx.html.Dimension.getBoxHeight( this._inputElement );
         if( inputElementHeight != 0 ) {
           if( qx.core.Variant.isSet( "qx.client", "mshtml" ) ) {
@@ -879,8 +879,10 @@
             top = 0;
           }
           top = Math.floor( top );
-          this._inputElement.style.position = "relative";
-          this._inputElement.style.top = top + "px";
+          // [if] Set padding instead of style.position of the _inputElement.
+          // style.position leads to problems with DOM events in FF 3.0.x
+          // see bug 292487 and bug 284356
+          this.setPaddingTop( top );
         }
       }
     },
