--- qx-build/source/class/qx/ui/form/TextField.js	2 Sep 2009 10:55:19 -0000	1.7
+++ qx-build/source/class/qx/ui/form/TextField.js	20 Oct 2009 09:56:39 -0000
@@ -879,8 +879,10 @@
             top = 0;
           }
           top = Math.floor( top );
-          this._inputElement.style.position = "relative";
-          this._inputElement.style.top = top + "px";
+          // [if] Set padding instead of style.position of the _inputElement.
+          // style.position leads to problems with DOM events in FF 3.0.x
+          // see bug 292487 and bug 284356
+          this.setStyleProperty( "paddingTop", top + "px" );
         }
       }
     },
