### Eclipse Workspace Patch 1.0
#P org.eclipse.rap.qx.build
Index: source/class/qx/ui/core/Parent.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/ui/core/Parent.js,v
retrieving revision 1.1
diff -u -r1.1 Parent.js
--- qx-build/source/class/qx/ui/core/Parent.js	10 Dec 2008 16:57:54 -0000	1.1
+++ qx-build/source/class/qx/ui/core/Parent.js	2 Nov 2009 20:32:49 -0000
@@ -212,7 +212,7 @@
         this.removeEventListener("keypress", this._onfocuskeyevent);
 
         // Deactivate focus handling
-        this.setTabIndex(-1);
+        this.setTabIndex(null);
 
         // Don't hide focus outline
         this.setHideFocus(false);
Index: source/class/qx/ui/core/Widget.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/ui/core/Widget.js,v
retrieving revision 1.6
diff -u -r1.6 Widget.js
--- qx-build/source/class/qx/ui/core/Widget.js	28 Oct 2009 13:54:27 -0000	1.6
+++ qx-build/source/class/qx/ui/core/Widget.js	2 Nov 2009 20:32:51 -0000
@@ -5936,7 +5936,7 @@
      * @return {var} TODOC
      */
     isFocusable : function() {
-      return this.getEnabled() && this.isSeeable() && this.getTabIndex() >= 0 && this.getTabIndex() != null;
+      return this.getEnabled() && this.isSeeable() && this.getTabIndex() != null;
     },
 
 
Index: source/class/qx/ui/form/Spinner.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/ui/form/Spinner.js,v
retrieving revision 1.3
diff -u -r1.3 Spinner.js
--- qx-build/source/class/qx/ui/form/Spinner.js	17 Jul 2009 14:48:33 -0000	1.3
+++ qx-build/source/class/qx/ui/form/Spinner.js	2 Nov 2009 20:32:53 -0000
@@ -119,7 +119,7 @@
     // ************************************************************************
     this._upbutton = new qx.ui.form.Button;
     this._upbutton.setAppearance("spinner-button-up");
-    this._upbutton.setTabIndex(-1);
+    this._upbutton.setTabIndex(null);
     this._upbutton.setHeight("1*");
     this._buttonlayout.add(this._upbutton);
 
@@ -128,7 +128,7 @@
     // ************************************************************************
     this._downbutton = new qx.ui.form.Button;
     this._downbutton.setAppearance("spinner-button-down");
-    this._downbutton.setTabIndex(-1);
+    this._downbutton.setTabIndex(null);
     this._downbutton.setHeight("1*");
     this._buttonlayout.add(this._downbutton);
 
Index: source/class/qx/ui/window/Window.js
===================================================================
RCS file: /cvsroot/rt/org.eclipse.rap/releng/org.eclipse.rap.qx.build/source/class/qx/ui/window/Window.js,v
retrieving revision 1.1
diff -u -r1.1 Window.js
--- qx-build/source/class/qx/ui/window/Window.js	10 Dec 2008 16:57:53 -0000	1.1
+++ qx-build/source/class/qx/ui/window/Window.js	2 Nov 2009 20:32:53 -0000
@@ -107,7 +107,7 @@
     var bm = this._minimizeButton = new qx.ui.form.Button;
 
     bm.setAppearance("window-captionbar-minimize-button");
-    bm.setTabIndex(-1);
+    bm.setTabIndex(null);
 
     bm.addEventListener("execute", this._onminimizebuttonclick, this);
     bm.addEventListener("mousedown", this._onbuttonmousedown, this);
@@ -120,7 +120,7 @@
     var br = this._restoreButton = new qx.ui.form.Button;
 
     br.setAppearance("window-captionbar-restore-button");
-    br.setTabIndex(-1);
+    br.setTabIndex(null);
 
     br.addEventListener("execute", this._onrestorebuttonclick, this);
     br.addEventListener("mousedown", this._onbuttonmousedown, this);
@@ -133,7 +133,7 @@
     var bx = this._maximizeButton = new qx.ui.form.Button;
 
     bx.setAppearance("window-captionbar-maximize-button");
-    bx.setTabIndex(-1);
+    bx.setTabIndex(null);
 
     bx.addEventListener("execute", this._onmaximizebuttonclick, this);
     bx.addEventListener("mousedown", this._onbuttonmousedown, this);
@@ -146,7 +146,7 @@
     var bc = this._closeButton = new qx.ui.form.Button;
 
     bc.setAppearance("window-captionbar-close-button");
-    bc.setTabIndex(-1);
+    bc.setTabIndex(null);
 
     bc.addEventListener("execute", this._onclosebuttonclick, this);
     bc.addEventListener("mousedown", this._onbuttonmousedown, this);
