--- qx-build/source/class/qx/client/History.js	27 Nov 2009 15:26:16 -0000	1.4
+++ qx-build/source/class/qx/client/History.js	27 Nov 2009 15:47:05 -0000
@@ -197,9 +197,11 @@
     {
       if (newTitle != null) {
         document.title = newTitle;
-        this._titles[state] = newTitle;
       }
+      this._titles[state] = document.title;
       if (state != this._state) {
+        // RAP [if] Prevent the event dispatch
+        this._state = state;
         // RAP [if] Fix for bug 295816
         //top.location.hash = "#" + encodeURIComponent(state)
         window.location.hash = "#" + encodeURIComponent(state);
@@ -271,7 +273,9 @@
 
       this._timer.addEventListener("interval", function(e) {
         var newHash = this.__getState();
-        if (newHash != this._state) {
+        // RAP [if] Ignore all non application states
+        //if (newHash != this._state) {
+        if( newHash != this._state && this._titles[newHash] != null ) {
           this.__onHistoryLoad(newHash);
         }
       }, this);
