--- qx-build/source/class/qx/ui/basic/ScrollBar.js	10 Dec 2008 16:57:43 -0000	1.1
+++ qx-build/source/class/qx/ui/basic/ScrollBar.js	26 Jan 2010 17:28:15 -0000
@@ -227,8 +227,16 @@
     {
       if (this._horizontal) {
         this._scrollContent.setWidth(value);
+        // fix for Bug 299620: set maximum and value in same call:
+        if( this._scrollContent.isCreated() ) {
+          this._scrollContent.getElement().style.width = value + "px";
+        }
       } else {
         this._scrollContent.setHeight(value);
+        // fix for Bug 299620: set maximum and value in same call:
+        if( this._scrollContent.isCreated() ) {
+          this._scrollContent.getElement().style.height = value + "px";
+        }
       }
 
       // recheck the value
@@ -369,6 +377,18 @@
     _afterAppear : function()
     {
       this.base(arguments);
+      // fix for RAP Bug 299620: scrolling in IE
+      if(    qx.core.Client.getEngine() == "mshtml" 
+          && this._scrollContent.isCreated() )
+      {
+        var style = this._scrollContent.getElement().style;
+        var width = style.width;
+        var height = style.height;
+        style.width = "0px";
+        style.height = "0px";
+        style.width = width;
+        style.height = height;       
+      } 
       this._positionKnob(this.getValue());
     }
   },
