--- qx-build/source/class/qx/event/handler/DragAndDropHandler.js	27 Nov 2009 20:35:26 -0000	1.4
+++ qx-build/source/class/qx/event/handler/DragAndDropHandler.js	2 Feb 2010 14:29:06 -0000
@@ -505,10 +505,11 @@
           if (this.__dragCache.dragHandlerActive)
           {
             // Fire first user events
-            this._fireUserEvents(this.__dragCache.currentDropWidget, this.__dragCache.sourceWidget, e);
+            var currentDropWidget = this.getDropTarget( e );
+            this._fireUserEvents( null, currentDropWidget, e);
 
             // Update status flags
-            this.__dragCache.currentDropWidget = this.__dragCache.sourceWidget;
+            this.__dragCache.currentDropWidget = currentDropWidget;
 
             // Activate capture for clientDocument
             qx.ui.core.ClientDocument.getInstance().setCapture(true);
