--- qx-build/source/class/qx/ui/core/Widget.js	1 Apr 2010 12:00:47 -0000	1.14
+++ qx-build/source/class/qx/ui/core/Widget.js	12 Apr 2010 10:26:03 -0000
@@ -3111,25 +3111,25 @@
     {
       if (value)
       {
-        if (this._isDisplayable) {
+        if ( this._isDisplayable && this._isCreated ) {
           this._beforeAppear();
         }
 
         this.removeStyleProperty("display");
 
-        if (this._isDisplayable) {
+        if ( this._isDisplayable && this._isCreated ) {
           this._afterAppear();
         }
       }
       else
       {
-        if (this._isDisplayable) {
+        if ( this._isDisplayable && this._isCreated ) {
           this._beforeDisappear();
         }
 
         this.setStyleProperty("display", "none");
 
-        if (this._isDisplayable) {
+        if ( this._isDisplayable && this._isCreated ) {
           this._afterDisappear();
         }
       }
