--- qx-build/source/class/qx/ui/form/TextField.js	2008-01-31 17:40:35.000000000 +0100
+++ qx-build/source/class/qx/ui/form/TextField.js	2008-07-21 17:31:39.000000000 +0200
@@ -1004,7 +1004,9 @@
       "default" : function(vStart)
       {
         this._visualPropertyCheck();
-        this._inputElement.selectionStart = vStart;
+        if( this.isSeeable() ) {
+          this._inputElement.selectionStart = vStart;
+        } 
       }
     }),
 
@@ -1042,7 +1044,7 @@
       "default" : function()
       {
         this._visualPropertyCheck();
-        return this._inputElement.selectionStart;
+        return this.isSeeable() ? this._inputElement.selectionStart : 0;
       }
     }),
 
@@ -1078,7 +1080,7 @@
 
         var el = this._inputElement;
 
-        if ( qx.util.Validation.isValidString(el.value) && this.getVisibility() ) {
+        if ( qx.util.Validation.isValidString(el.value) && this.isSeeable() ) {
           el.selectionEnd = el.selectionStart + vLength;
         }
       }
@@ -1112,7 +1114,7 @@
         this._visualPropertyCheck();
 
         var el = this._inputElement;
-        return el.selectionEnd - el.selectionStart;
+        return this.isSeeable() ? ( el.selectionEnd - el.selectionStart ) : 0;
       }
     }),
 
@@ -1150,6 +1152,9 @@
       "default" : function(vText)
       {
         this._visualPropertyCheck();
+        if( !this.isSeeable() ) {
+          return;
+        }
 
         var el = this._inputElement;
 
@@ -1244,10 +1249,11 @@
       "default" : function(vStart, vEnd)
       {
         this._visualPropertyCheck();
-
-        var el = this._inputElement;
-        el.selectionStart = vStart;
-        el.selectionEnd = vEnd;
+        if( this.isSeeable() ) {
+          var el = this._inputElement;
+          el.selectionStart = vStart;
+          el.selectionEnd = vEnd;
+        }
       }
     })
   },
