--- qx-build/source/class/qx/ui/tree/TreeFolder.js	2008-01-31 17:40:32.000000000 +0100
+++ qx-build/source/class/qx/ui/tree/TreeFolder.js	2008-07-21 18:08:34.000000000 +0200
@@ -933,6 +933,7 @@
             this.getTree().getManager().handleMouseUp(this, e);
             e._treeProcessed = true;
           }
+          this.getTree().getManager().handleMouseUp(e.getTarget(), e);
       }
     },
 
